Sources/PointFree/Episode/Episode.swift

```diff
@@ -9,50 +9,40 @@ import Optics
 import Prelude
 import Styleguide
 
-let episodeResponse: Middleware<StatusLineOpen, ResponseEnded, Either<String, Int>, Data> =
-  fetchEpisode
-    >-> responseEpisode
-
-private func fetchEpisode(_ conn: Conn<StatusLineOpen, Either<String, Int>>) -> IO<Conn<HeadersOpen, Episode?>> {
+let episodeResponse =
+  map(episode(for:))
+    >>> (
+      requireSome(notFoundView: episodeNotFoundView)
+        <| requestContextMiddleware
+        >-> writeStatus(.ok)
+        >-> respond(
+          episodeView.map(addHighlightJs >>> addGoogleAnalytics)
+      )
+)
 
-  let possibleEpisode = episodes.first(where: {
-    conn.data.left == .some($0.slug)
-      || conn.data.right == .some($0.id)
+private func episode(for param: Either<String, Int>) -> Episode? {
+  return episodes.first(where: {
+    param.left == .some($0.slug) || param.right == .some($0.id)
   })
-
-  return conn.map(const(possibleEpisode))
-    |> writeStatus(possibleEpisode == nil ? .notFound : .ok)
 }
 
-private func responseEpisode(_ conn: Conn<HeadersOpen, Episode?>) -> IO<Conn<ResponseEnded, Data>> {
-
-  switch conn.data {
-  case .none:
-    return conn.map(const(unit))
-      |> respond(notFoundView)
-  case let .some(ep):
-    return conn.map(const(ep))
-      |> respond(episodeView.map(addHighlightJs >>> addGoogleAnalytics))
-  }
-}
-
-public let episodeView = View<Episode> { ep in
+let episodeView = View<RequestContext<Episode>> { context in
   document([
     html([
       head([
         style(renderedNormalizeCss),
         style(styleguide),
-        title("Episode #\(ep.sequence): \(ep.title)"),
+        title("Episode #\(context.data.sequence): \(context.data.title)"),
         meta(viewport: .width(.deviceWidth), .initialScale(1)),
         ]),
 
       body(
         [`class`([Class.pf.colors.bg.dark])],
-        navView.view(unit) <> [
+        navView.view(context.map(const(unit))) <> [
```

Comment:
will soon figure out how to make a layout view that takes care of header and footer automatically.